### PR TITLE
Refactoring to use pojo for metadata

### PR DESCRIPTION
**Pull request by galactoise**

### CODE REVIEW

File: voice-interface/src/main/java/com/derpgroup/derpwizard/voice/model/SsmlDocumentBuilder.java [hunk 1]
```diff
@@ -82,6 +82,7 @@ public String toString() {
   private Set<String> ignoreTags;
   private List<List<StringBuilder>> paragraphs;
   private int index = 0;
+  private boolean conversationEnd = true;
```
Comment (davidwphillips) [inline voice-interface/src/main/java/com/derpgroup/derpwizard/voice/model/SsmlDocumentBuilder.java:85]: This seems like the wrong place for conversation status things.

File: voice-interface/src/main/java/com/derpgroup/derpwizard/voice/model/SsmlDocumentBuilder.java [hunk 2]
```diff
@@ -249,6 +250,15 @@ public String toString() {
     return buildString();
   }
 
+  public boolean isConversationEnd() {
```
Comment (R-Gerard) [inline voice-interface/src/main/java/com/derpgroup/derpwizard/voice/model/SsmlDocumentBuilder.java:253]: What's the purpose of this method?

Comment (galactoise) [inline voice-interface/src/main/java/com/derpgroup/derpwizard/voice/model/SsmlDocumentBuilder.java:253]: To tell the implementing resource whether it should treat this as the final response in the conversation.

File: voice-interface/src/main/java/com/derpgroup/derpwizard/voice/model/SsmlDocumentBuilder.java (hunk 1 above)
Comment (galactoise) [inline voice-interface/src/main/java/com/derpgroup/derpwizard/voice/model/SsmlDocumentBuilder.java:85]: I happen to agree.  We just don't really have a better place for it, other than maybe stashing it in the metadata.

Comment (R-Gerard) [inline voice-interface/src/main/java/com/derpgroup/derpwizard/voice/model/SsmlDocumentBuilder.java:85]: I think making this a method in the CommonMetadata class makes more sense.

Comment (galactoise) [inline voice-interface/src/main/java/com/derpgroup/derpwizard/voice/model/SsmlDocumentBuilder.java:85]: I'm sort of leaning that way, as well.  This code was done before CommonMetadata existed.  The question will be how to deal with serialization of that field.

Anyhow, I think that can be addressed with the later PR where the code is more settled.

**Pull request merged**